internal/autometrics/ctx.go

```diff
@@ -28,13 +29,106 @@ type GeneratorContext struct {
 	//
 	// This can be set in the command for the generator or through the environment.
 	DisableDocGeneration bool
+	// ImportMap maps the alias to import in the current file, to canonical names associated with that name.
+	ImportsMap map[string]string
+}
+
+// This is almost a carbon copy of the autometrics.Context structure, except that
+// non-literal types are transcribed to strings to make it possible to reason about
+// the runtime context within generator logic.
+// Also, all the information that only gets filled at runtime is simply ignored, as
+// only statically-known information is useful at this point.
+type RuntimeCtxInfo struct {
+	// Name of the variable to use as context.Context when building the autometrics.Context.
+	// The string will be empty iff 'nil' must be used as autometrics.NewContext() argument.
+	ContextVariableName string
+	// Verbatim code to use to fetch the TraceID.
+	// For example, if the instrumented function is detected to use Gin, like
+	// `func ginHandler(ginVarName *gin.Context)`
+	// then the code in that variable should be something like
+	// `"autometrics.DecodeString(ginVarName.GetString(\"TraceID\"))"`
+	// This getter should return []byte to allow PreInstrument to
+	// log warnings and fill the data manually if the getter returns nil.
+	TraceIDGetter        string
+	SpanIDGetter         string
+	TrackConcurrentCalls bool
+	TrackCallerName      bool
+	AlertConf            *autometrics.AlertConfiguration
+}
+
+func DefaultRuntimeCtxInfo() RuntimeCtxInfo {
+	return RuntimeCtxInfo{
+		TrackConcurrentCalls: true,
+		TrackCallerName:      true,
+		ContextVariableName:  "nil",
+	}
+}
+
+func (c RuntimeCtxInfo) Validate(allowCustomLatencies bool) error {
+	if c.AlertConf != nil {
+		if c.AlertConf.ServiceName == "" {
+			return fmt.Errorf("Cannot have an AlertConfiguration without a service name")
```

Comment:
all errors here who are just a static string (without usage of format variables) can just use smth like `errors.New`

Reply:
I knew there must have been something for that :)